clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java

```diff
@@ -5634,16 +5637,31 @@ public void testListOffsetsMetadataNonRetriableErrors() throws Exception {
 
         final TopicPartition tp1 = new TopicPartition("foo", 0);
 
-        try (AdminClientUnitTestEnv env = new AdminClientUnitTestEnv(cluster)) {
-            env.kafkaClient().setNodeApiVersions(NodeApiVersions.create());
+        Map<MetadataResponse, Class<? extends Throwable>> responsesAndFailures = new HashMap<>();
```

Comment:
extremely minor nit (we could do in a followup and not a blocker bug) but could we parameterize this?

Reply:
Leave it cleaner than you found it!

Pushed a commit that adds parameterization; if you'd prefer, can revert and publish as a follow-up PR after we merge this one.